src/main/kotlin/org/rust/ide/console/RsConsoleRunner.kt

```diff
@@ -62,17 +69,80 @@ class RsConsoleRunner(project: Project) :
         return consoleView
     }
 
-    override fun fillToolBarActions(
-        toolbarActions: DefaultActionGroup,
+
+
+    override fun createContentDescriptorAndActions() {
```

Comment:
This code looks like a bad copy-paste of an old Java code 👎 

Please consider something like this:

```Kotlin
override fun createContentDescriptorAndActions() {
    val outputActions = listOf(
        SoftWrapAction(consoleView),
        ScrollToTheEndToolbarAction(consoleView.editor),
        PrintAction(consoleView)
    )
    val actionManager = ActionManager.getInstance()

    val runActionGroup = DefaultActionGroup()
    val runToolbar = actionManager.createActionToolbar("RustConsoleRunner", runActionGroup, false)

    val outputActionGroup = DefaultActionGroup().apply {
        addAll(outputActions)
    }
    val outputToolbar = actionManager.createActionToolbar("RustConsoleRunner", outputActionGroup, false).apply {
        val emptyBorderSize = component.border.getBorderInsets(component).left
        val outsideBorder = BorderFactory.createMatteBorder(0, 1, 0, 0, JBColor.border())
        val insideBorder = JBEmptyBorder(emptyBorderSize)
        component.border = BorderFactory.createCompoundBorder(outsideBorder, insideBorder)
    }

    val actionsPanel = JPanel(BorderLayout()).apply {
        add(runToolbar.component, BorderLayout.WEST)
        add(outputToolbar.component, BorderLayout.CENTER)
    }

    val mainPanel = JPanel(BorderLayout()).apply {
        add(actionsPanel, BorderLayout.WEST)
        add(consoleView.component, BorderLayout.CENTER)
        runToolbar.setTargetComponent(this)
        outputToolbar.setTargetComponent(this)
    }

    val title = constructConsoleTitle(consoleTitle)
    val contentDescriptor = RunContentDescriptor(consoleView, processHandler, mainPanel, title, consoleIcon).apply {
        setFocusComputable { consoleView.consoleEditor.contentComponent }
        isAutoFocusContent = isAutoFocusContent
        Disposer.register(project, this)
    }

    val runActions = listOf(
        RestartAction(this),
        createConsoleExecAction(consoleExecuteActionHandler),
        StopAction(processHandler!!),
        CloseAction(executor, contentDescriptor, project)
    )
    runActionGroup.addAll(runActions)
    val actions = runActions + outputActions + EOFAction()

    registerActionShortcuts(actions, consoleView.consoleEditor.component)
    registerActionShortcuts(actions, mainPanel)

    showConsole(executor, contentDescriptor)
}
```

Reply:
Yes, actually I took this code from Python console. Thank you, code with `apply` is much cleaner